src/main/java/seedu/recipe/ui/RecipeCard.java

```diff
@@ -160,4 +241,60 @@ public boolean equals(Object other) {
         return id.getText().equals(card.id.getText())
                 && recipe.equals(card.recipe);
     }
+
+    private boolean showConfirmationDialog() {
+        Stage confirmationDialog = new Stage();
+        confirmationDialog.initModality(Modality.APPLICATION_MODAL);
+        confirmationDialog.setTitle("Confirm Deletion");
+
+        AtomicBoolean confirmed = new AtomicBoolean(false);
+
+        Label label = new Label("Are you sure you want to delete this recipe?");
+        Button confirmButton = new Button("Confirm");
+        Button cancelButton = new Button("Cancel");
+        // set confirm button on action
+        confirmButton.setOnAction(e -> {
+            confirmed.set(true);
+            confirmationDialog.close();
+        });
+        // set cancel button on action
+        cancelButton.setOnAction(e -> confirmationDialog.close());
+        // set confirm button on key pressed
+        confirmButton.setOnKeyPressed(event -> {
+            if (event.getCode() == KeyCode.ENTER) {
+                confirmed.set(true);
+                confirmationDialog.close();
+            }
+        });
+        // set cancel button on key pressed
+        cancelButton.setOnKeyPressed(event -> {
+            if (event.getCode() == KeyCode.ENTER) {
+                confirmed.set(false);
+                confirmationDialog.close();
+            }
+        });
+        // initialize HBox buttons
+        HBox buttons = new HBox(10);
+        buttons.getChildren().addAll(confirmButton, cancelButton);
+        buttons.setAlignment(Pos.CENTER);
+        // initialize VBox layout
+        VBox layout = new VBox(10);
+        layout.getChildren().addAll(label, buttons);
+        layout.setAlignment(Pos.CENTER);
```

Comment:
again, not super sure how UI works but you may want to abstract out this part into a `ConfirmationDialog` component?